apis/python/src/tiledbsoma/_index_util.py

```diff
@@ -28,20 +20,10 @@ def tiledbsoma_build_index(
     Args:
         keys:
            Integer keys used to build the index (hash) table.
-       context:
-           ``SOMATileDBContext`` object containing concurrecy level (exclusive with thread_count).
-       thread_count:
-           Concurrency level when the user does not want to use ``context``.
 
     Lifecycle:
         Experimental.
     """
-    if context is not None:
-        tdb_concurrency = int(
-            context.tiledb_ctx.config().get("sm.compute_concurrency_level", 10)
-        )
-        thread_count = max(1, tdb_concurrency // 2)
```

Comment:
I'm looking at this again more closely, and what you can do instead is 
`tdb_concurrency = int(context.tiledb_config.get("sm.compute_concurrency_level", 10))`

Using `context.tiledb_ctx` will create a new context if it doesn't exist. Using `context.tiledb_config` will not. If you don't actually need to use the context and all you need to do is get this setting out of the config then just use the latter.